src/esp/agent/Agent.h

```diff
@@ -112,6 +112,10 @@ class Agent : public Magnum::SceneGraph::AbstractFeature3D {
 
   void setState(const AgentState& state, const bool resetSensors = true);
 
+  void setSensorSuite(sensor::SensorSuite sensorSuite) {
+    sensors_ = sensorSuite;
```

Comment:
This argument should be passed by reference, instead of making a copy.

Reply:
So : 

`void setSensorSuite(const sensor::SensorSuite& sensorSuite)`

Reply:
Shouldn't this be `sensor::SensorSuite&& sensorSuite` as the arg and then `sensors_ = std::move(sensorSuite);`?  `sensors_` is an instance of sensor::SenorSuite, so the assignment will make a copy currently

Reply:
I considered mentioning that, but I wasn't sure if we would want to retain the sensorSuite we pass to this function in the calling code.  If that is not a concern, then the move would probably be a better choice.

Reply:
@bigbike Thoughts?

Reply:
I think @erikwijmans made a very good point here. Nice catch! We should directly move it instead of copying.
Actually I would even consider deleting the copy constructor in the sensorSuite.

Reply:
Thank you! That's fixed now.